spec.bs

```diff
@@ -1419,6 +1454,60 @@ To <dfn>validate and convert auction ad config</dfn> given an {{AuctionAdConfig}
   1. If |buyer| is failure, then return failure.
   1. [=map/Set=] |auctionConfig|'s [=auction config/per buyer priority signals=][|buyer|] to
     |signals|.
+1. If |config|["{{AuctionAdConfig/auctionReportBuyerKeys}}"] [=map/exists=]:
+  1. Let |interestGroupBuyers| be |auctionConfig|'s [=auction config/interest group buyers=].
+  1. If |interestGroupBuyers| is null, set |interestGroupBuyers| to a new [=list=].
+  1. [=list/For each=] |index| of [=the exclusive range|the range=] 0 to |config|[
+    "{{AuctionAdConfig/auctionReportBuyerKeys}}"]'s [=list/size=], exclusive:
+    1. Let |key| be |config|["{{AuctionAdConfig/auctionReportBuyerKeys}}"][|index|].
+    1. If |key| is not [=set/contained=] in [=the exclusive range|the range=] 0 to 2<sup>128</sup>,
+      exclusive, [=exception/throw=] a {{TypeError}}.
+    1. If |index| is equal to or greater than |interestGroupBuyers|' [=list/size=],
+      [=iteration/continue=].
+
+        Note: [=iteration/Continue=] is used (instead of [=iteration/break=]) to match validation
+        logic for all given buyer keys.
+    1. Let |origin| be |interestGroupBuyers|[|index|].
+    1. [=map/Set=] |auctionConfig|'s [=auction config/auction report buyer keys=][|origin|]
+      to |key|.
+
+    Issue: Check behavior when an origin is repeated in {{AuctionAdConfig/interestGroupBuyers}}.
+1. If |config|["{{AuctionAdConfig/auctionReportBuyers}}"] [=map/exists=]:
+  1. [=map/For each=] |reportType| → |reportBuyerConfig| of |config|[
+    "{{AuctionAdConfig/auctionReportBuyers}}"]:
+    1. If « "`interestGroupCount`", "`bidCount`", "`totalGenerateBidLatency`",
+      "`totalSignalsFetchLatency`" » does not [=list/contain=] |reportType|, [=iteration/continue=].
+
+      Note: No error is thrown to allow forward compatibility if
+          additional report types are added later.
+
+      Issue: Should these strings be dash delimited?
+
+    1. If |reportBuyerConfig|["{{AuctionReportBuyersConfig/bucket}}"] is not [=set/contained=] in
+      [=the exclusive range|the range=] 0 to 2<sup>128</sup>, exclusive, [=exception/throw=] a
+      {{TypeError}}.
+
+      Issue: Consider validating the case where the bucket used (after summing) is too large.
+      Currently, the implementation appears to overflow. See
+      <a href="https://github.com/WICG/turtledove/issues/1040">protected-audience/1040</a>.

```

Comment:
nit: Update to match other issues, i.e. `(<a href="https://github.com/WICG/turtledove/issues/1040">WICG/turtledove#1040</a>)`

Reply:
Done